submariner/OVN-Interconnect.md

```diff
@@ -0,0 +1,244 @@
+# Submariner Enhancement for OVN Interconnect
+
+<!-- Add link to issue/epic if available -->
+
+## Summary
+
+OVN CNI is moving to a new network topology powered by OVN Interconnect Feature. This allows independent OVN deployments,
+within a kubernetes cluster, to be interconnected by OVN-managed GENEVE tunnels. Submariner does not work with these changes
+out of the box and this proposal explains the changes required to support it.
+
+## Proposal
+
+With OVN Interconnect we can have two types of deployment
+
+* Single Zone(Global Zone): A single-zone deployment will have only one OVN database and a set of master nodes programming it. This is similar
+  to the topology that we use now. But it will have a single `global` zone assigned to it with a transit switch which is not involved in any
+  packet forwarding. Though this setup is expected to work by default, the subnet that the transit switch uses overlaps with the subnet
+  Submariner uses for the logical router. In this proposal, we plan to remove the submariner logical router and the associated switches and
+  this will not be a problem anymore.
+
+  The below indicates node annotations in a single zone OVN setup and is added by OVN.
+
+```bash
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.1/16"]'
+      k8s.ovn.org/ovn-zone: global
+    name: cluster1-worker
+
+    annotations:
+      k8s.ovn.org/ovn-node-transit-switch-port-ips: '["169.254.0.1/16"]'
+      k8s.ovn.org/ovn-zone: global
```

Comment:
When I deployed a OVN KIND cluster with IC enabled, I see the following annotations.

```
k8s.ovn.org/node-transit-switch-port-ifaddr: {"ipv4":"168.254.0.3/16"}                                                                                                       
k8s.ovn.org/zone-name: ovn-worker 
```

So, I looked at the code and see the above mentioned annotations.
https://github.com/ovn-org/ovn-kubernetes/blob/346360757b89b689d2ca2cb090698be7d0be1419/go-controller/pkg/util/node_annotations.go#L89

Which is the right annotation?

Reply:
this should be the latest, I took those from a PR which was under review.